lib/core/src/Cardano/Pool/Metadata.hs

```diff
@@ -194,31 +198,59 @@ defaultRecordTypeOptions = Aeson.defaultOptions
 getStakePoolMetadata
     :: Trace IO RegistryLog
     -- ^ Logging object - use 'transformTrace' to convert to 'Text'.
+    -> FilePath
+    -- ^ Directory to cache the metadata registry.
     -> String
     -- ^ URL of a zip archive to fetch and extract.
     -> [PoolOwner]
     -- ^ List of stake pools to get metadata for.
     -> IO (Either FetchError [Maybe StakePoolMetadata])
-getStakePoolMetadata tr url poolOwners = fmap join $ tryJust fileExceptionHandler $
-    withSystemTempFile "stake-pool-metadata.zip" $ \zipFileName zipFile -> do
-        let tr' = contramap (fmap (RegistryLog url zipFileName)) tr
-        logTrace tr' MsgDownloadStarted
-        fetchStakePoolMetaZip url zipFile >>= \case
-            Right size -> do
-                logTrace tr' $ MsgDownloadComplete size
-                hClose zipFile
-                Right <$> getMetadataFromZip tr' zipFileName poolOwners
-            Left e -> pure $ Left e
+getStakePoolMetadata tr cacheDir url poolOwners = do
+    let zipFileName = cacheDir </> "registry.zip"
+    let tr' = contramap (fmap (RegistryLog url zipFileName)) tr
+    createDirectoryIfMissing False cacheDir
+    fetchStakePoolMetaZipCached tr' url zipFileName >>= \case
+        Right _ -> Right <$> getMetadataFromZip tr' zipFileName poolOwners
+        Left e -> pure $ Left e
+
+cacheTTL :: NominalDiffTime
+cacheTTL = 3600
+
+fetchStakePoolMetaZipCached
+    :: Trace IO RegistryLogMsg
+    -> String
+    -> FilePath
+    -> IO (Either FetchError ())
+fetchStakePoolMetaZipCached tr url zipFileName = checkCached >>= \case
+    Just mtime ->
+        Right <$> logTrace tr (MsgUsingCached zipFileName mtime)
+    Nothing ->
+       fetchStakePoolMetaZip tr url zipFileName
   where
-    fileExceptionHandler = Just . FetchErrorFile . displayException @IOException
-
--- | Fetch a URL, streaming to the given file handle.
-fetchStakePoolMetaZip :: String -> Handle -> IO (Either FetchError Integer)
-fetchStakePoolMetaZip url zipFile = tryJust handler $ do
-    req <- parseUrlThrow url
-    httpSink req $ \_response -> Conduit.mapM_ (BS.hPut zipFile)
-    hTell zipFile
+    checkCached = try (getModificationTime zipFileName) >>= \case
+        Right mtime -> do
+            now <- getCurrentTime
+            pure $ if (diffUTCTime now mtime < cacheTTL)
+                then Just mtime
+                else Nothing
+        Left (_ :: IOException) -> pure Nothing
```

Comment:
**[question]**

I suppose that `getModificationTime` throws an IOException when the file doesn't exist?